### PR TITLE
Generate static forwarders for object members in companion interface

### PR DESCRIPTION
**Pull request by retronym**
We used to disable generation of static forwarders when a object had a
trait as a companion, as one could not add methods with bodies to an
interface in JVM 6.

The JVM lifted this restriction to support default methods in interfaces,
so we can lift the restriction on static forwarders, too.

Fixes https://github.com/scala/scala-dev/issues/59

Review by @lrytz 

/cc @ktoso

### CODE REVIEW

**Comment by ktoso** [comment]
This is awesome, thanks a lot @retronym (from the entire team btw)! :-)

**Comment by lrytz** [comment]
the commit message talks about `GenASM`, which might confuse some future-us.

File: src/compiler/scala/tools/nsc/backend/jvm/BCodeSkelBuilder.scala [hunk 1]
```diff
@@ -164,7 +164,7 @@ abstract class BCodeSkelBuilder extends BCodeHelpers {
 
       } else {
 
-        val skipStaticForwarders = (claszSymbol.isInterface || settings.noForwarders)
+        val skipStaticForwarders = settings.noForwarders
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/backend/jvm/BCodeSkelBuilder.scala]: inline this boolean?

**Comment by retronym** [comment]
@lrytz sorry about the delay, I've just addressed your comments.

**Comment by lrytz** [comment]
LGTM!

**Pull request merged**